src/credentials/CHIPCert.h

```diff
@@ -604,83 +605,44 @@ CHIP_ERROR DetermineCertType(ChipCertificateData & cert);
+ *   Convert a CHIP epoch time into an ASN.1 universal time structure.
  *
- * @param packedTime  A packed certificate time to be unpacked.
- * @param time        A reference to an ASN1UniversalTime structure to receive the unpacked date/time.
+ * @param epochTime  A CHIP epoch time to be converted.
+ * @param asn1Time   A reference to an ASN1UniversalTime structure to receive the date/time.
  *
- * @retval  #CHIP_NO_ERROR                      If the input time was successfully unpacked.
+ * @retval  #CHIP_NO_ERROR                      If the input time was successfully converted.
  */
-CHIP_ERROR UnpackCertTime(uint32_t packedTime, chip::ASN1::ASN1UniversalTime & time);
-
-/**
- * @brief
- *   Convert a packed certificate date/time to a packed certificate date.
- *
- * @details
- *   A packed certificate date contains the fields of a calendar date--year, month, day--packed into an
- *   unsigned integer.  The bits are organized such that ordinal comparisons of packed date values
- *   correspond to the natural ordering of the corresponding dates.  To reduce their size, packed
- *   certificate dates are limited to representing dates on or after 2020/01/01.  When housed within
- *   a 16-bit unsigned integer, packed certificate dates can represent dates up to the year 2196.
- *
- * @param packedTime  The packed certificate date/time to be converted.
- *
- * @return A corresponding packed certificate date.
- **/
-uint16_t PackedCertTimeToDate(uint32_t packedTime);
-
-/**
- * @brief
- *   Convert a packed certificate date to a corresponding packed certificate date/time, where
- *   the time portion of the value is set to 00:00:00.
- *
- * @param packedDate  The packed certificate date to be converted.
- *
- * @return  A corresponding packed certificate date/time.
- **/
-uint32_t PackedCertDateToTime(uint16_t packedDate);
+CHIP_ERROR ChipEpochToASN1Time(uint32_t epochTime, chip::ASN1::ASN1UniversalTime & asn1Time);
 
 /**
  * @brief
- *   Convert the number of seconds since 1970-01-01 00:00:00 UTC to a packed certificate date/time.
+ *   Convert the number of seconds since 1970-01-01 00:00:00 UTC to a CHIP Epoch time.
  *
- * @param secondsSinceEpoch  Number of seconds since 1970-01-01 00:00:00 UTC.
- *                           Note: this value is compatible with *positive* values
- *                           of the POSIX time_t value, up to the year 2105.
+ * @param unixEpoch  Number of seconds since 1970-01-01 00:00:00 UTC.
+ *                   Note: this value is compatible with *positive* values
+ *                   of the POSIX time_t value, up to the year 2105.
+ * @param chipEpoch  Number of seconds since 2020-01-01 00:00:00 UTC.
  *
- * @return  A corresponding packed certificate date/time.
+ * @return  #CHIP_NO_ERROR  If the input was successfully converted.
  **/
-uint32_t SecondsSinceEpochToPackedCertTime(uint32_t secondsSinceEpoch);
+CHIP_ERROR UnixEpochToChipEpochTime(uint32_t unixEpoch, uint32_t & chipEpoch);
```

Comment:
We already have this in `TimeUtils`.  In fact, since we are declaring it here but not defining anywhere around here we are relying on `TimeUtils` being linked in, no?  Why are we not including that header here if we want this?  Or, since nothing in this diff uses this function, why can't we remove this declaration?

Reply:
Good Catch! This is redundant here. We will be using the one defined in `TimeUtils`.